File: Week-4/HW-1/Drivers/Driver.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;


public class Driver
{
    private Driver()
    {
    }

    public static IWebDriver driver;

    public static WebDriver get()
    {
        if (driver == null)
        {
            driver = new ChromeDriver();
        }
        return (WebDriver)driver;
    }

    public static void closeDriver()
    {
        if (driver != null)
        {
            driver.Quit();
            driver = null;
        }
    }
}

File: Week-5/HW-2/src/test/java/Pinterest/utilities/Driver.java
package Pinterest.utilities;

import io.appium.java_client.MobileElement;
import io.appium.java_client.android.AndroidDriver;
import org.openqa.selenium.remote.DesiredCapabilities;

import java.net.MalformedURLException;
import java.net.URL;
import java.util.concurrent.TimeUnit;

public class Driver {

    private static AndroidDriver driver;

    public static AndroidDriver getDriver() throws MalformedURLException {
        if (driver == null) {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.setCapability("platformName", "Android");
            caps.setCapability("deviceName", "emulator-5554");
            caps.setCapability("fullReset", "true");
            caps.setCapability("noReset", "false");
            caps.setCapability("autoGrantPermissions", "true");
            caps.setCapability("app", System.getProperty("user.dir") + "/bin/com.pinterest.apk");

            driver = new AndroidDriver<MobileElement>(new URL("http://127.0.0.1:4723/wd/hub"), caps);
            driver.manage().timeouts().implicitlyWait(20, TimeUnit.SECONDS);
        }
        return driver;
    }

    public static void closeDriver() {
        if (driver != null) {
            driver.quit();
            driver = null;
        }
    }
}